### PR TITLE
Implement T-CONV and T-REPLACE-RX entry placeholders

### LINKED ISSUE #7293: Add support for {T-CONV:/Text/Type/} and {T-REPLACE-RX:/.../.../} as placeholder in fields
Following up on #1825, please consider adding support for {T-CONV:/Text/Type/} and {T-REPLACE-RX:/.../.../} as placeholder in fields. It's currently the feature I miss the most while switching over from KeePass.

My use-case:
Im trying to pass a base64-encoded value to a powershell script via the URL field:
`cmd://cmd /C "start powershell -ExecutionPolicy ByPass -File ^"{DB_DIR}\scripts\Test.ps1^" -Base64 {T-CONV:/{USERNAME}/Base64/}"`

The suggested workaround of manually converting the required value and storing it in a custom field does not work for us. We use field references (e.g. Username: `{REF:U@I:46C9B1FFBD4ABC4BBB260C6190BAD20C}`) and those cannot be used to retrieve custom fields.

### PR DESCRIPTION
**Pull request by droidmonkey**
* Closes #7293
* Move existing T-CONV and T-REPLACE-RX code from AutoType to Entry. Replumb AutoType to use the entry functions.
* Improve placeholder code in various place

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Added unit tests

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

### CODE REVIEW

File: tests/TestEntry.cpp [hunk 1]
```diff
@@ -514,6 +514,68 @@ void TestEntry::testResolveNonIdPlaceholdersToUuid()
...
+    entry2->setUrl("cmd://ssh {USERNAME}@server.com -p {PASSWORD}");
+
+    // Test complicated and nested conversions
+    QCOMPARE(entry1->resolveMultiplePlaceholders(entry1->title()), QString("Title1 username1 PASSWORD1"));
+    QCOMPARE(entry2->resolveMultiplePlaceholders(entry2->url()),
+             QString("cmd://ssh USERNAME1@server.com -p Password1"));
+    // Test base64 and hex conversions
+    QCOMPARE(entry1->resolveMultiplePlaceholders("{T-CONV:/{PASSWORD}/base64/}"), QString("UGFzc3dvcmQx"));
+    QCOMPARE(entry1->resolveMultiplePlaceholders("{T-CONV:/{PASSWORD}/hex/}"), QString("50617373776f726431"));
+    // Test URL encode and decode
```
Comment (varjolintu) [inline tests/TestEntry.cpp:545]: Should the tests include invalid conversion as well? Or with some invalid string/placeholder?

Comment (droidmonkey) [reply tests/TestEntry.cpp:545]: Yes good point, technically those cases are covered by the autotype tests.

Comment (droidmonkey) [reply tests/TestEntry.cpp:545]: @varjolintu all done

**Review by varjolintu — APPROVED** [review]
Works great. Tried it also with invalid values and ignored the UI warnings to check if the application complains about them when launching Auto-Type or opening a cmd URL.

**Pull request merged**